internal/grpc/services/datatx/datatx.go

```diff
@@ -81,20 +156,228 @@ func (s *service) UnprotectedEndpoints() []string {
 	return []string{}
 }
 
-func (s *service) CreateTransfer(ctx context.Context, req *datatx.CreateTransferRequest) (*datatx.CreateTransferResponse, error) {
-	return &datatx.CreateTransferResponse{
-		Status: status.NewUnimplemented(ctx, errtypes.NotSupported("CreateTransfer not implemented"), "CreateTransfer not implemented"),
-	}, nil
+func (s *service) PullTransfer(ctx context.Context, req *datatx.PullTransferRequest) (*datatx.PullTransferResponse, error) {
+	srcEp, err := s.extractEndpointInfo(ctx, req.SrcTargetUri)
+	if err != nil {
+		return nil, err
+	}
+	srcRemote := fmt.Sprintf("%s://%s", srcEp.endpointScheme, srcEp.endpoint)
+	srcPath := srcEp.filePath
+	srcToken := srcEp.token
+
+	destEp, err := s.extractEndpointInfo(ctx, req.DestTargetUri)
+	if err != nil {
+		return nil, err
+	}
+	dstRemote := fmt.Sprintf("%s://%s", destEp.endpointScheme, destEp.endpoint)
+	dstPath := destEp.filePath
+	dstToken := destEp.token
+
+	txInfo, startTransferErr := s.txManager.StartTransfer(ctx, srcRemote, srcPath, srcToken, dstRemote, dstPath, dstToken)
+
+	// we always save the transfer regardless of start transfer outcome
+	// only then, if starting fails, can we try to restart it
+	txShare := &txShare{
+		TxID:          txInfo.GetId().OpaqueId,
+		SrcTargetURI:  req.SrcTargetUri,
+		DestTargetURI: req.DestTargetUri,
+		Opaque:        req.Opaque,
+	}
+	s.txShareDriver.Lock()
+	defer s.txShareDriver.Unlock()
+
+	s.txShareDriver.model.TxShares[txInfo.GetId().OpaqueId] = txShare
+	if err := s.txShareDriver.model.saveTxShare(); err != nil {
+		err = errors.Wrap(err, "datatx service: error saving transfer share: "+datatx.Status_STATUS_INVALID.String())
+		return &datatx.PullTransferResponse{
+			Status: status.NewInvalid(ctx, "error pulling transfer"),
+		}, err
```

Comment:
@ishank011 : I wonder, what are the criteria for returning the error vs. setting the response status error ?